File: Practica 3/Cola/Cola.py
class Cola:
    def _init_(self, n):
        self.__arreglo = [None] * n
        self.__inicio = 0
        self.__fin = 0
        self.__n = n

    def insert(self, e):
        if self.isFull():
            print("Cola llena")
        else:
            self._arreglo[self._fin] = e
            self.__fin += 1

    def remove(self):
        if self.isEmpty():
            print("Cola vacía")
            return None
        else:
            e = self._arreglo[self._inicio]
            self.__inicio += 1
            return e

    def peek(self):
        if self.isEmpty():
            print("Cola vacía")
            return None
        else:
            return self._arreglo[self._inicio]

    def isEmpty(self):
        return self._inicio == self._fin

    def isFull(self):
        return self._fin == self._n

    def size(self):
        return self._fin - self._inicio

File: Practica 3/Cola/Cola.java
package Cola;

public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = 0;
    }

    public void insert(long e) {
        if (fin == n) {
            System.out.println("Cola llena");
        } else {
            arreglo[fin] = e;
            fin++;
        }
    }

    public long remove() {
        if (inicio == fin) {
            System.out.println("Cola vacía");
            return -1;
        } else {
            long e = arreglo[inicio];
            inicio++;
            return e;
        }
    }

    public long peek() {
        if (inicio == fin) {
            System.out.println("Cola vacía");
            return -1;
        }
        return arreglo[inicio];
    }

    public boolean isEmpty() {
        return inicio == fin;
    }

    public boolean isFull() {
        return fin == n;
    }

    public int size() {
        return fin - inicio;
    }
}